conans/client/build/meson.py

```diff
@@ -194,21 +192,47 @@ def build(self, args=None, build_dir=None, targets=None):
         ])
         self._run("ninja %s" % arg_list)
 
+    def _run_meson_command(self, subcommand=None, args=None, build_dir=None):
+        args = args or []
+        build_dir = build_dir or self.build_dir or self._conanfile.build_folder
+
+        arg_list = join_arguments([
+            subcommand,
+            '-C "%s"' % build_dir,
+            args_to_string(args)
+        ])
+        self._run("meson %s" % arg_list)
+
+    def build(self, args=None, build_dir=None, targets=None):
+        if not self._conanfile.should_build:
+            return
+        self._run_ninja_targets(args=args, build_dir=build_dir, targets=targets)
+
     def install(self, args=None, build_dir=None):
         if not self._conanfile.should_install:
             return
         mkdir(self._conanfile.package_folder)
         if not self.options.get('prefix'):
             raise ConanException("'prefix' not defined for 'meson.install()'\n"
                                  "Make sure 'package_folder' is defined")
-        self.build(args=args, build_dir=build_dir, targets=["install"])
+        self._run_ninja_targets(args=args, build_dir=build_dir, targets=["install"])
 
     def test(self, args=None, build_dir=None, targets=None):
         if not self._conanfile.should_test:
             return
         if not targets:
             targets = ["test"]
-        self.build(args=args, build_dir=build_dir, targets=targets)
+        self._run_ninja_targets(args=args, build_dir=build_dir, targets=targets)
+
+    def meson_install(self, args=None, build_dir=None):
+        if not self._conanfile.should_install:
+            return
+        self._run_meson_command(subcommand='install', args=args, build_dir=build_dir)
```

Comment:
These commands might not be supported by all meson versions, so @TheQwertiest is suggesting to add a version check to decide behavior. I am fine with that.